pymongo/asynchronous/command_cursor.py

```diff
@@ -381,21 +385,28 @@ async def __aenter__(self) -> AsyncCommandCursor[_DocumentType]:
     async def __aexit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
         await self.close()
 
-    async def to_list(self) -> list[_DocumentType]:
+    async def to_list(self, length: int = -1) -> list[_DocumentType]:
         """Converts the contents of this cursor to a list more efficiently than ``[doc async for doc in cursor]``.
 
         To use::
 
           >>> await cursor.to_list()
 
+        Or, so read at most n items from the cursor::
+
+          >>> await cursor.to_list(n)
+
         If the cursor is empty or has no more results, an empty list will be returned.
 
         .. versionadded:: 4.9
         """
         res: list[_DocumentType] = []
+        remaining = length
         while self.alive:
-            if not await self._next_batch(res):
+            if remaining != 0 and not await self._next_batch(res, remaining):
                 break
+            if length != -1:
+                remaining = length - len(res)
```

Comment:
In this case, I think someone can pass `-2` and it be valid. Should we also throw a `ValueError` here as well?

Reply:
I'm going to switch to Optional[int], per Shane's recommendation.